File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.c
int* rearrangeArray(int* arr, int n, int* r) {
    int p=0;
    int ne=1;
    int *res = malloc(n * sizeof(int));
    for (int i=0;i<n;i++){
        if(arr[i]<0){
            res[ne]=arr[i];
            ne+=2;
        }
        else{
            res[p]=arr[i];
            p+=2;
        }
    }
    *r=n;
    return res;

}

File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.py
class Solution(object):
    def rearrangeArray(self, arr):
        n=len(arr)
        p=[]
        ne=[]
        for i in range(n):
            if(arr[i]<0):
                ne.append(arr[i])
            else:
                p.append(arr[i])
        res=[]
        for i in range(n/2):
            res.append(p[i])
            res.append(ne[i])
        return res        